lib/node_modules/@stdlib/math/base/napi/ternary/src/fif_f.c

```diff
@@ -0,0 +1,98 @@
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/napi/ternary/fif_f.h"
+#include <node_api.h>
+#include <stdint.h>
+#include <assert.h>
+
+/**
+* Invokes a ternary function accepting a single-precision floating-point number, a signed 32-bit integer, and a single-precision floating-point number and returning a single-precision floating-point number.
+*
+* ## Notes
+*
+* -   This function expects that the callback `info` argument provides access to the following JavaScript arguments:
+*
+*     -   `x`: input value.
+*     -   `y`: input value.
+*     -   `z`: input value.
+*
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @param fcn    ternary function
+* @return       function return value as a Node-API double-precision floating-point number
+*/
+napi_value stdlib_math_base_napi_fif_f( napi_env env, napi_callback_info info, float (*fcn)( float, int32_t, float ) ) {
+	napi_status status;
+
+	size_t argc = 3;
+	napi_value argv[ 3 ];
+	status = napi_get_cb_info( env, info, &argc, argv, NULL, NULL );
+	assert( status == napi_ok );
+
+	if ( argc < 3 ) {
+		status = napi_throw_error( env, NULL, "invalid invocation. Must provide three numbers." );
+		assert( status == napi_ok );
+		return NULL;
+	}
+
+	napi_valuetype vtype0;
+	status = napi_typeof( env, argv[ 0 ], &vtype0 );
+	assert( status == napi_ok );
+	if ( vtype0 != napi_number ) {
+		status = napi_throw_type_error( env, NULL, "invalid argument. First argument must be a number." );
+		assert( status == napi_ok );
+		return NULL;
+	}
+
+	napi_valuetype vtype1;
+	status = napi_typeof( env, argv[ 1 ], &vtype1 );
+	assert( status == napi_ok );
+	if ( vtype1 != napi_number ) {
+		status = napi_throw_type_error( env, NULL, "invalid argument. Second argument must be a number." );
+		assert( status == napi_ok );
+		return NULL;
+	}
+
+	napi_valuetype vtype2;
+	status = napi_typeof( env, argv[ 2 ], &vtype2 );
+	assert( status == napi_ok );
+	if ( vtype2 != napi_number ) {
+		status = napi_throw_type_error( env, NULL, "invalid argument. Third argument must be a number." );
+		assert( status == napi_ok );
+		return NULL;
+	}
+
+	double x;
+	status = napi_get_value_double( env, argv[ 0 ], &x );
+	assert( status == napi_ok );
+
+	int32_t y;
+	status = napi_get_value_int32( env, argv[ 1 ], &y );
+	assert( status == napi_ok );
+
+	double z;
+	status = napi_get_value_double( env, argv[ 2 ], &z );
+	assert( status == napi_ok );
+
+	napi_value v;
+	status = napi_create_double( env, (double)fcn( (float)x, (int32_t)y, (float)z ), &v );
```

Comment:
```suggestion
	status = napi_create_double( env, (double)fcn( (float)x, y, (float)z ), &v );
```

`y` already is `int32_t`. No need to cast again here.